src/components/container.js

```diff
@@ -279,27 +278,46 @@ function createFunctionContainer({ displayName, override } = {}) {
       ? store === override.Store
       : store.containedBy === FunctionContainer;
 
-  function FunctionContainer({ children, scope, isGlobal, ...restProps }) {
+  function FunctionContainer(props) {
+    const { children, ...restProps } = props;
+    const { scope, isGlobal, ...subProps } = restProps;
     const ctx = useContext(Context);
     const registry = useRegistry(scope, isGlobal, ctx);
+
+    // Store props in a ref to avoid re-binding actions when they change and re-rendering all
+    // consumers unnecessarily. The update is handled by an effect on the component instead
+    const propsRef = useRef({ prev: null, next: restProps, sub: subProps });
+    propsRef.current = {
+      prev: propsRef.current.next,
+      next: restProps,
+      sub: subProps, // TODO remove on next major
```

Comment:
To properly trigger onContainerUpdate we need to have 2 versions of props:
- one with scope to ensure we trigger the callback on scope change
- one without to maintain compat with store actions API
We also add prev props as it is now cheap and was asked in #131

Reply:
Why can we remove it on the next version?

Reply:
Because I'd like to ship a breaking change: passing `scope` and `isGlobal` too. Right now we filter them out and just return custom props. But having scope available can be handy sometimes